pkg/volume/csi/csi_attacher.go

```diff
@@ -228,23 +228,6 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		return errors.New(log("attacher.MountDevice failed, deviceMountPath is empty"))
 	}
 
-	corruptedDir := false
-	mounted, err := isDirMounted(c.plugin, deviceMountPath)
-	if err != nil {
-		klog.Error(log("attacher.MountDevice failed while checking mount status for dir [%s]", deviceMountPath))
-		if isCorruptedDir(deviceMountPath) {
-			corruptedDir = true // leave to CSI driver to handle corrupted mount
-			klog.Warning(log("attacher.MountDevice detected corrupted mount for dir [%s]", deviceMountPath))
-		} else {
-			return err
-		}
-	}
-
-	if mounted && !corruptedDir {
```

Comment:
I realized this could be a breaking change for lots of CSI drivers, hold for discussion
/hold

Reply:
If the directory is already mounted and not corrupt  shouldn't we return successfully ?

Reply:
Okay, got it - we offload all of the path related checks (of the mount directories only) to the CSI drivers ?